go-controller/pkg/ovn/gress_policy.go

```diff
@@ -166,30 +141,39 @@ func (gp *gressPolicy) constructPeerASMatchString(v4AddressSets, v6AddressSets [
 	//  At this point there will be address sets in one or both of them.
 	//  Contents in both address sets mean dual stack, else one will be empty because we will only populate
 	//  entries for enabled stacks
-	if len(v4AddressSets) > 0 {
+	if config.IPv4Mode {
 		v4AddressSetStr := strings.Join(v4AddressSets, ", ")
-		v4MatchStr = fmt.Sprintf("%s.%s == {%s}", "ip4", direction, v4AddressSetStr)
+		v4Match = fmt.Sprintf("%s.%s == {%s}", "ip4", direction, v4AddressSetStr)
```

Comment:
"Previously l3Match for gress with namespace selector that doesn't select
anything looked like "ip4.src == {<empty address set>}", now it will be
"ip4.src == {}"."

Do we really want to program ACLs that wont match on anything? In the previous behavior it made sense because address sets are populated without touching ACL again. But ACL will have to be updated to have the right address set once the selector finally matches something.

Reply:
What is the behavior here of ovn-controller when the match is an empty brace? Does it throw any error or warning, does it assume it does not match anything? @dceara ?

Reply:
ovn-controller complains, e.g.:
```
2023-03-15T11:06:40.219Z|00030|lflow|WARN|error parsing match "ip4.src == {}": Syntax error at `}' expecting constant.
```
So the flow will not be instsalled (and will obviously never match).

Reply:
ok i will use the update instead of mutate for port group acl updates then

Reply:
A bit of extra clarification: the `"ip4.src == {}"` match can be replaced by `"0"` with the exact same effect - nothing will match.